WordPress/Classes/ViewRelated/NUX/SignupUsernameTableViewController.swift

```diff
@@ -0,0 +1,237 @@
+                return
+        }
+
+        suggestUsernames(for: nameToSearch) { [weak self] (suggestions) in
+            self?.suggestions = suggestions
+            self?.tableView.reloadSections(IndexSet(integersIn: Sections.searchField.rawValue...Sections.suggestions.rawValue), with: .automatic)
+        }
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        SVProgressHUD.dismiss()
+    }
+
+    private func suggestUsernames(for searchTerm: String, addSuggestions: @escaping ([String]) ->()) {
+        guard !isSearching else {
+            return
+        }
+
+        isSearching = true
+
+        let context = ContextManager.sharedInstance().mainContext
+        let accountService = AccountService(managedObjectContext: context)
+        guard let account = accountService.defaultWordPressComAccount(),
+            let api = account.wordPressComRestApi else {
+            return
+        }
+        SVProgressHUD.show(withStatus: NSLocalizedString("Loading usernames", comment: "Shown while the app waits for the username suggestions web service to return during the site creation process."))
+
+        let service = AccountSettingsService(userID: account.userID.intValue, api: api)
+        service.suggestUsernames(base: searchTerm) { [weak self] (newSuggestions) in
+            self?.isSearching = false
+            SVProgressHUD.dismiss()
+            addSuggestions(newSuggestions)
+        }
+    }
+
+    func setupBackgroundTapGestureRecognizer() {
+        let gestureRecognizer = UITapGestureRecognizer()
+        gestureRecognizer.on(call: { [weak self] (gesture) in
+            self?.view.endEditing(true)
+        })
+        gestureRecognizer.cancelsTouchesInView = false
+        view.addGestureRecognizer(gestureRecognizer)
+    }
+}
+
+// MARK: UITableViewDataSource
+
+extension SignupUsernameTableViewController {
+    fileprivate enum Sections: Int {
+        case titleAndDescription = 0
+        case searchField = 1
+        case suggestions = 2
+
+        static var count: Int {
+            return suggestions.rawValue + 1
+        }
+    }
+
+    private enum SuggestionStyles {
+        static let indentationWidth: CGFloat = 20.0
+        static let indentationLevel = 1
+    }
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return Sections.count
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case Sections.titleAndDescription.rawValue:
+            return 1
+        case Sections.searchField.rawValue:
+            return suggestions.count > 0 ? 1 : 0
+        case Sections.suggestions.rawValue:
+            return suggestions.count
+        default:
+            return 0
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
```

Comment:
Per the design, the current username should be the top row and pre-selected:
<img width="385" alt="screen shot 2018-02-25 at 11 24 53 am" src="https://user-images.githubusercontent.com/1816888/36644927-8ba2cd32-1a1e-11e8-9d1b-c5e1a9089683.png">

<img width="285" alt="screen shot 2018-02-25 at 11 25 15 am" src="https://user-images.githubusercontent.com/1816888/36644929-9427c4b2-1a1e-11e8-812e-3a1edddc9fe9.png">

Reply:
Ah, I missed that detail. Nice catch!